src/main/kotlin/app/revanced/patches/twitter/misc/fxtwitter/FxTwitterPatch.kt

```diff
@@ -0,0 +1,134 @@
+
+            for ((index, instruction) in instructions.withIndex()) {
+                if (instruction.opcode == Opcode.CONST_STRING) {
+                    val string = ((instruction as ReferenceInstruction).reference as StringReference).string
+
+                    if (string.contains("https://twitter.com/%1\$s/status/%2\$d") || string.contains("https://x.com/%1\$s/status/%2\$d")) {
+                        val overrideRegister = (instruction as OneRegisterInstruction).registerA
+
+                        var overrideString = string
+                        if (string.contains("twitter.com")) {
+                            overrideString = string.replace("twitter.com", shareLinkBase.toString())
+                        } else if (string.contains("x.com")) {
+                            overrideString = string.replace("x.com", shareLinkBase.toString())
+                        }
+
+                        this.mutableMethod.replaceInstruction(
+                            index,
+                            """
+                                const-string v$overrideRegister, "$overrideString"
+                            """
+                        )
+                    }
+                }
+            }
+        } ?: throw PatchException("FxTwitterPatchFingerprint not found")
+
+
+        FxTwitterPatchFingerprintResource.result?.apply {
+            val instructions = this.mutableMethod.getInstructions()
+
+            // Save user nickname to v9
+            for ((index, instruction) in instructions.withIndex()) {
+                if (instruction.opcode == Opcode.INVOKE_VIRTUAL) {
+
+                    val methodRef =
+                        (instruction as ReferenceInstruction).reference as MethodReference
+                    if (
+                        methodRef.returnType != "Ljava/lang/String;") continue;
+
+                    val sourceRegister = (instructions[index + 1] as OneRegisterInstruction).registerA
+
+                    this.mutableMethod.addInstruction(
+                        index + 2,
+                        """
+                            move-object v9, v$sourceRegister
+                        """
+                    )
+
+                    break;
+                }
+            }
+
+            // Save tweet id to v7, v8
+            for ((index, instruction) in instructions.withIndex()) {
+                if (instruction.opcode == Opcode.INVOKE_VIRTUAL) {
+
+                    val methodRef =
+                        (instruction as ReferenceInstruction).reference as MethodReference
+                    if (methodRef.definingClass != "Ljava/lang/Long;") continue;
+
+                    val sourceRegister = (instructions[index + 1] as OneRegisterInstruction).registerA
+
+                    this.mutableMethod.addInstruction(
+                        index + 2,
+                        """
+                            move-wide v7, v$sourceRegister
```

Comment:
Hardcoding the register can fail, you should get the register from the instruction and reuse it here.